libraries/AP_RangeFinder/RangeFinder_Backend.h

```diff
@@ -80,8 +79,12 @@ class AP_RangeFinder_Backend
     void set_status(RangeFinder::RangeFinder_Status status);
 
     RangeFinder::RangeFinder_State &state;
-    MAV_DISTANCE_SENSOR sensor_type;
 
     // semaphore for access to shared frontend data
     AP_HAL::Semaphore *_sem;    
+
+    virtual MAV_DISTANCE_SENSOR _get_mav_distance_sensor_type() const {
+        // most of the backends are lasers:
+        return MAV_DISTANCE_SENSOR_LASER;
```

Comment:
I wasn't saying to make it pure-virtual (although I'm also fine with that), I was saying that the backend definition should return the general case, which is UNKNOWN. Having it return LASER by default would mean that any driver that didn't override it would return LASER - and that could be wrong. It's very irrelevant in this case since nothing will happen if the wrong sensor type is returned, but it's about the principle.

Anyway, I'm fine with either pure-virtual or having a backend definition that returns UNKNOWN, your choice. This is based on #6737, so I'll only merge this after that one, giving you some time to decide 🙂